### PR TITLE
1222: Git patch parser fails on Status T

### PR DESCRIPTION
**Pull request by zhaosongzs**
Make parser know Status T

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-1222](https://bugs.openjdk.org/browse/SKARA-1222): Git patch parser fails on Status T

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to [e9500b37](https://git.openjdk.org/skara/pull/1373/files/e9500b37252017bf14a19def1f81adf74e3b9b5b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara pull/1373/head:pull/1373` \
`$ git checkout pull/1373`

Update a local copy of the PR: \
`$ git checkout pull/1373` \
`$ git pull https://git.openjdk.org/skara pull/1373/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1373`

View PR using the GUI difftool: \
`$ git pr show -t 1373`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1373.diff">https://git.openjdk.org/skara/pull/1373.diff</a>

</details>

### CODE REVIEW

File: vcs/src/main/java/org/openjdk/skara/vcs/tools/PatchHeader.java [hunk 1]
```diff
@@ -103,6 +103,9 @@ public static PatchHeader fromRawLine(String line) {
             sourcePath = Path.of(parts[1]);
         } else if (status.isUnmerged()) {
             sourcePath = Path.of(parts[1]);
+        } else if (status.isFileTypeChanged()) {
+            sourcePath = Path.of(parts[1]);
+            targetPath = sourcePath;
```
Comment (zhaosongzs) [inline vcs/src/main/java/org/openjdk/skara/vcs/tools/PatchHeader.java:108]: I am not very sure whether this is right or not. I could not create a Status T raw line by myself

**Comment by lgxbslgx** [comment]
Please enable the GHA in your fork [1] and run it munually [2]. So as other PRs. Note: You don't need to run the GHA munually in the future, but GitHub can't help you run the GHA of your previous PRs.

[1] https://wiki.openjdk.org/display/SKARA/Testing
[2] https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow

**Comment by zhaosongzs** [comment]
> 
Sure, I will run the tests tomorrow. Thanks for your guidance!


> Please enable the GHA in your fork [1] and run it munually [2]. So as other PRs. Note: You don't need to run the GHA munually in the future, but GitHub can't help you run the GHA of your previous PRs.
> 
> [1] https://wiki.openjdk.org/display/SKARA/Testing [2] https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow

**Comment by magicus** [comment]
You can also just push an empty dummy commit after enabling GHA to trigger a run; that might be easier.

**Review by erikj79 — APPROVED** [review]

**Review by magicus — APPROVED** [review]
Looks good to me.

**Comment by zhaosongzs** [comment]
> You can also just push an empty dummy commit after enabling GHA to trigger a run; that might be easier.

Yes, it's easy for me, thanks for your advice!

**Comment by erikj79** [comment]
Tests are passing so you can post `/integrate` now.

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**